tests/update/tests.py

```diff
@@ -138,3 +140,49 @@ def test_update_respects_to_field(self):
         self.assertEqual(bar_qs[0].foo_id, a_foo.target)
         bar_qs.update(foo=b_foo)
         self.assertEqual(bar_qs[0].foo_id, b_foo.target)
+
+    def test_update_annotated_queryset(self):
+        """
+        Update of a queryset which has been annotated.
+        """
+        # Trivial annotated update first
+        qs = DataPoint.objects.annotate(alias=F('value'))
+        updated = qs.update(another_value='foo')
+        self.assertEqual(updated, 3)
+        # Update where annotation is used for filtering
+        qs = DataPoint.objects.annotate(alias=F('value'))
+        qs = qs.filter(alias='apple')
+        updated = qs.update(another_value='foo')
+        self.assertEqual(updated, 1)
+        # Update where annotation is used in update parameters
+        qs = DataPoint.objects.annotate(alias=F('value'))
+        updated = qs.update(another_value=F('alias'))
+        self.assertEqual(updated, 3)
+        # Update where aggregation annotation is used in update parameters
+        qs = DataPoint.objects.annotate(max=Max('value'))
+        with self.assertRaises(FieldError):
```

Comment:
use `with self.assertRaisesMessage(FieldError, "Aggregate functions are't allowed in this query."):`